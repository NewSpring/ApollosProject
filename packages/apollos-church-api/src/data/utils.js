import ApollosConfig from '@apollosproject/config';

const { ROCK, ROCK_MAPPINGS } = ApollosConfig;

export const createAssetUrl = (value) =>
  value.path
    ? value.path
    : `${ROCK.IMAGE_URL}/${
        ROCK_MAPPINGS.ASSET_STORAGE_PROVIDERS[`${value.AssetStorageProviderId}`]
      }/${value.Key}`;

export const fakeFunction = () => 'Fake';
