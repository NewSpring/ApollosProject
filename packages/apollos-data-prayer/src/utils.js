import ApollosConfig from '@apollosproject/config';
import { AuthenticationError } from 'apollo-server';
import jwt from 'jsonwebtoken';

export const secret = process.env.SECRET || 'ASea$2gadj#asd0';

export const parseToken = (token) => jwt.verify(token, secret);

export const registerToken = (token) => {
  try {
    const { cookie, sessionId } = parseToken(token);

    return {
      userToken: token,
      rockCookie: cookie,
      sessionId,
    };
  } catch (e) {
    throw new AuthenticationError('Invalid token');
  }
};

export const generateToken = (params) =>
  jwt.sign({ ...params }, secret, { expiresIn: '60d' });

export const enforceProtocol = (uri) =>
  uri.startsWith('//') ? `https:${uri}` : uri;

export const createImageUrlFromGuid = (uri) =>
  uri.split('-').length === 5
    ? `${ApollosConfig.ROCK.IMAGE_URL}?guid=${uri}`
    : enforceProtocol(uri);

export const latLonDistance = (lat1, lon1, lat2, lon2) => {
  if (lat1 === lat2 && lon1 === lon2) {
    return 0;
  }
  const radlat1 = (Math.PI * lat1) / 180;
  const radlat2 = (Math.PI * lat2) / 180;
  const theta = lon1 - lon2;
  const radtheta = (Math.PI * theta) / 180;
  let dist =
    Math.sin(radlat1) * Math.sin(radlat2) +
    Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta);
  if (dist > 1) {
    dist = 1;
  }
  dist = Math.acos(dist);
  dist = (dist * 180) / Math.PI;
  dist = dist * 60 * 1.1515;
  return dist;
};
