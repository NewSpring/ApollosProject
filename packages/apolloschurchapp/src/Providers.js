import React from 'react';
import { Providers } from '@apollosproject/ui-kit';
import ClientProvider from './client';

import theme from './theme'; // where theme looks like:

const NewSpringProviders = (props) => (
  <Providers themeInput={theme}>
    <ClientProvider {...props} />
  </Providers>
);

export default NewSpringProviders;
