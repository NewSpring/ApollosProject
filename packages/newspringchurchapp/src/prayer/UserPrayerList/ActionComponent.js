import React from 'react';
import ActionSheet from 'react-native-actionsheet';
import PropTypes from 'prop-types';
import { Icon, Touchable, withTheme } from '@apollosproject/ui-kit';

const DeleteIcon = withTheme(({ theme }) => ({
  name: 'close',
  size: theme.sizing.baseUnit,
  fill: theme.colors.wordOfChrist,
}))(Icon);

class ActionComponent extends React.Component {
  handleShowActionSheet = () => {
    this.ActionSheet.show();
  };

  render() {
    const { options } = this.props;
    return (
      <>
        <Touchable onPress={this.handleShowActionSheet}>
          <DeleteIcon />
        </Touchable>
        <ActionSheet
          ref={(sheet) => {
            this.ActionSheet = sheet;
          }}
          options={options.map((option) => option.title)}
          cancelButtonIndex={options.length - 1}
          destructiveButtonIndex={options
            .map((option) => option.destructive)
            .indexOf(true)}
          onPress={async (index) => {
            if (index === 0) {
              options[0].method();
            }
          }}
        />
      </>
    );
  }
}

ActionComponent.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({})),
};

export default ActionComponent;
