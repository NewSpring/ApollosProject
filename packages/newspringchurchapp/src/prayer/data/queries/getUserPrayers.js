import gql from 'graphql-tag';

export default gql`
  query UserPrayers {
    userPrayers {
      id
      firstName
      lastName
      text
      enteredDateTime
    }
  }
`;
