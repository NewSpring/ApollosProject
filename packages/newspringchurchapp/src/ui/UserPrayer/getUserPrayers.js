import gql from 'graphql-tag';

export default gql`
  query {
    getCurrentPersonPrayerRequests {
      id
      firstName
      lastName
      text
      enteredDateTime
    }
  }
`;
