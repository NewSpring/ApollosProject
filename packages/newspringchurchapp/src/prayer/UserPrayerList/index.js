import React from 'react';
import { Mutation, Query } from 'react-apollo';
import { View, ScrollView } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import {
  Card,
  CardContent,
  H4,
  H6,
  ModalView,
  PaddedView,
  styled,
} from '@apollosproject/ui-kit';
import PrayerSingle from '../PrayerSingle';
import GET_USER_PRAYERS from '../data/queries/getUserPrayers';
import DELETE_PRAYER from '../data/mutations/deletePrayer';
import ActionComponent from './ActionComponent';

const FlexedSafeAreaView = styled({
  flex: 1,
})(SafeAreaView);

const StyledView = styled(({ theme }) => ({
  marginBottom: theme.sizing.baseUnit * 4,
}))(View);

const GreenH4 = styled(({ theme }) => ({
  color: theme.colors.primary,
}))(H4);

class UserPrayerList extends React.Component {
  static navigationOptions = {
    header: null,
  };

  render() {
    return (
      <ModalView {...this.props} onClose={() => this.props.navigation.pop()}>
        <FlexedSafeAreaView forceInset={{ top: 'always' }}>
          <ScrollView>
            <Query query={GET_USER_PRAYERS} fetchPolicy="cache-and-network">
              {({ data: { userPrayers = [] } = {} }) => (
                <>
                  <PaddedView>
                    <H6>Viewing</H6>
                    <GreenH4>My Prayers</GreenH4>
                  </PaddedView>
                  <Mutation
                    mutation={DELETE_PRAYER}
                    update={(cache, { data: { deletePrayer } }) => {
                      const data = cache.readQuery({
                        query: GET_USER_PRAYERS,
                      });
                      const { id } = deletePrayer;
                      const updatedPrayers = data.userPrayers.filter(
                        (prayer) => prayer.id !== id
                      );
                      cache.writeQuery({
                        query: GET_USER_PRAYERS,
                        data: { userPrayers: updatedPrayers },
                      });
                    }}
                  >
                    {(deletePrayer) => (
                      <StyledView>
                        {userPrayers.map((prayer) => (
                          <Card key={prayer.id}>
                            <CardContent>
                              <PrayerSingle
                                prayer={prayer}
                                showDate
                                action={
                                  <ActionComponent
                                    options={[
                                      {
                                        title: 'Delete Prayer',
                                        method: async () => {
                                          await deletePrayer({
                                            variables: {
                                              parsedId: prayer.id,
                                            },
                                          });
                                        },
                                        destructive: true,
                                      },
                                      {
                                        title: 'Cancel',
                                        method: null,
                                        destructive: false,
                                      },
                                    ]}
                                  />
                                }
                              />
                            </CardContent>
                          </Card>
                        ))}
                      </StyledView>
                    )}
                  </Mutation>
                </>
              )}
            </Query>
          </ScrollView>
        </FlexedSafeAreaView>
      </ModalView>
    );
  }
}

export default UserPrayerList;
