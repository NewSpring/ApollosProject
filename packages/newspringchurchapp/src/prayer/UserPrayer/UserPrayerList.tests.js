import React from 'react';
import { renderWithApolloData } from 'newspringchurchapp/src/utils/testUtils';

import Providers from 'newspringchurchapp/src/Providers';
import getUserPrayers from './getUserPrayers';
import UserPrayerList from '.';

const prayers = [
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4786',
    firstName: 'Aaron',
    lastName: 'Attendee',
    text: 'This is a test to see if I have a prayer request.',
    enteredDateTime: '2017-04-08T11:21:53.053',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4787',
    firstName: 'Aaron',
    lastName: 'Attendee',
    text: 'This is another test to see if I have a prayer request.',
    enteredDateTime: '2017-04-09T11:21:53.053',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4788',
    firstName: 'Aaron',
    lastName: 'Attendee',
    text: 'This is just another test to see if I have a prayer request.',
    enteredDateTime: '2017-04-11T11:21:53.053',
  },
  {
    id: 'PrayerRequest:b99bbd5002a0bf8d4c73e86afe9d4789',
    firstName: 'Aaron',
    lastName: 'Attendee',
    text: 'This, yet again, is a test to see if I have a prayer request.',
    enteredDateTime: '2017-04-10T11:21:53.053',
  },
];

describe('the UserPrayerList component', () => {
  it('renders a list of prayers', async () => {
    const mock = {
      request: {
        query: getUserPrayers,
      },
      response: {
        data: {
          prayers,
        },
      },
    };
    const tree = await renderWithApolloData(
      <Providers mocks={[mock]} addTypename={false}>
        <UserPrayerList />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
