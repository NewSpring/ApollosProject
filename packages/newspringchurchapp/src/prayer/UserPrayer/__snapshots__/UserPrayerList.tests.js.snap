// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the UserPrayerList component renders a list of prayers 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 64,
          "marginTop": 24,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
